domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/validation/genericlayout/GenericLayoutShowConditionValidatorImpl.java

```diff
@@ -0,0 +1,101 @@
+        this.caseFieldEntityUtil = caseFieldEntityUtil;
+    }
+
+    @Override
+    public ValidationResult validate(GenericLayoutEntity entity, List<GenericLayoutEntity> allGenericLayouts) {
+        ValidationResult validationResult = new ValidationResult();
+        Optional<String> showConditionOptional = entity.fetchShowCondition();
+
+        if (!showConditionOptional.isPresent()) {
+            return validationResult;
+        }
+
+        List<GenericLayoutEntity> layoutEntities = getLayoutsByMatchingCaseType(entity, allGenericLayouts);
+
+        ShowCondition showCondition;
+        try {
+            showCondition = showConditionExtractor.parseShowCondition(showConditionOptional.get());
+        } catch (InvalidShowConditionException e) {
+            validationResult.addError(
+                new ValidationError(String.format(ERROR_MESSAGE_SHOW_CONDITION, showConditionOptional.get(),
+                        entity.getCaseType().getReference(), entity.getCaseField().getReference()), entity));
+            return validationResult;
+        }
+
+        List<String> allSubTypePossibilities = getAllSubTypePossibilities(layoutEntities);
+        showCondition.getFieldsWithSubtypes().forEach(showConditionField -> {
+            if (!allSubTypePossibilities.contains(showConditionField)) {
+                validationResult.addError(buildError(entity, showConditionField, showConditionOptional.get()));
+            }
+        });
+
+        showCondition.getFields().forEach(showConditionField -> {
+            if (!forShowConditionFieldExistsAtLeastOneEventCaseFieldEntity(
+                showConditionField, layoutEntities)
+                && !MetadataField.isMetadataField(showConditionField)) {
+                validationResult.addError(buildError(entity, showConditionField, showConditionOptional.get()));
+            }
+        });
+
+        return validationResult;
+    }
+
+    private List<String> getAllSubTypePossibilities(List<GenericLayoutEntity> layoutEntities) {
+        return caseFieldEntityUtil.buildDottedComplexFieldPossibilities(
+                layoutEntities.stream()
+                    .map(GenericLayoutEntity::getCaseField)
+                    .collect(Collectors.toList()));
+    }
+
+    private List<GenericLayoutEntity> getLayoutsByMatchingCaseType(GenericLayoutEntity entity,
+                                                                   List<GenericLayoutEntity> allGenericLayouts) {
+        return allGenericLayouts.stream()
+            .filter(l -> l.getCaseType().getReference().equalsIgnoreCase(entity.getCaseType().getReference()))
+            .collect(Collectors.toList());
+    }
+
+    private boolean forShowConditionFieldExistsAtLeastOneEventCaseFieldEntity(
+        String showConditionField, List<GenericLayoutEntity> layoutEntities) {
+        return layoutEntities
+            .stream()
+            .anyMatch(f -> f.getCaseField().getReference().equals(showConditionField));
```

Comment:
If find `f` a bit misleading. Can we rename to `le` for layout entity?

Reply:
done